native/src/main/java/io/ballerina/lib/avro/serialize/visitor/SerializeVisitor.java

```diff
@@ -95,32 +98,32 @@ private Object serializeField(Schema schema, Object fieldData) throws Exception
 
     @Override
     public Object visit(PrimitiveSerializer primitiveSerializer, Object data) throws Exception {
-        switch (primitiveSerializer.getSchema().getType()) {
-            case INT -> {
-                return ((Long) data).intValue();
-            }
-            case FLOAT -> {
-                return ((Double) data).floatValue();
+        return switch (primitiveSerializer.getSchema().getType()) {
+            case INT -> ((Long) data).intValue();
+            case FLOAT -> ((Double) data).floatValue();
+            case DOUBLE -> {
+                if (data instanceof Long longValue) {
```

Comment:
Lets add cases for `int`, `byte` types to map with ballerina integer subtypes.

Reply:
Added in [a05d61b](https://github.com/ballerina-platform/module-ballerina-avro/pull/27/commits/a05d61bfa3e4bb399001436e5df0cefe9bf2b20b)